pkg/kubelet/dockertools/container_labels.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockertools
+
+import (
+	"strconv"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/types"
+)
+
+const (
+	kubernetesPodNameLabel      = "io.kubernetes.pod.name"
+	kubernetesPodNamespaceLabel = "io.kubernetes.pod.namespace"
+
+	kubernetesPodLabel                    = "io.kubernetes.pod.data"
```

Comment:
I read code about this yesterday, it seems that one of the options is to add some "better-formatted" labels on each container (rather than add the whole pod data), for example add a label like "container_metadata" and write all needed data like PreStop handler into it...